{
  "uid" : "411682f0177cfaf8",
  "name" : "test_insufficient_approval",
  "fullName" : "tests.brownie.token.tests.test_transferFrom#test_insufficient_approval",
  "historyId" : "0ae63a08126a8f0616f222dcfe6ba70d",
  "time" : {
    "start" : 1642074001840,
    "stop" : 1642074001840,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f116f627520>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
  "statusTrace" : "brownie_project = <Project 'TokenProject'>\naccounts = <brownie.network.account.Accounts object at 0x7f405705bdc0>\n\n    @pytest.fixture(scope=\"module\")\n    def token(brownie_project, accounts):\n>       return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n\ntests/brownie/token/tests/conftest.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py:600: in __call__\n    return tx[\"from\"].deploy(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:508: in deploy\n    receipt, exc = self._make_transaction(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:725: in _make_transaction\n    raise VirtualMachineError(e) from None\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = VirtualMachineError(ValueError(\"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries ex...ion refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"))\nexc = \"Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by N...ction refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\"\n\n    def __init__(self, exc: ValueError) -> None:\n        try:\n            exc = exc.args[0]\n        except Exception:\n            pass\n    \n        if isinstance(exc, dict) and \"message\" in exc:\n            if \"data\" not in exc:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.message: str = exc[\"message\"].rstrip(\".\")\n    \n            if isinstance(exc[\"data\"], str):\n                # handle parity exceptions - this logic probably is not perfect\n                if \"0x08c379a0\" in exc[\"data\"]:\n                    revert_type, err_msg = [i.strip() for i in exc[\"data\"].split(\"0x08c379a0\", 1)]\n                    err_msg = eth_abi.decode_abi([\"string\"], HexBytes(err_msg))\n                    err_msg = f\"{revert_type} '{err_msg}'\"\n                elif exc[\"data\"].endswith(\"0x\"):\n                    err_msg = exc[\"data\"][:-2].strip()\n                else:\n                    err_msg = exc[\"data\"]\n                raise ValueError(f\"{self.message}: {err_msg}\") from None\n    \n            try:\n                txid, data = next((k, v) for k, v in exc[\"data\"].items() if k.startswith(\"0x\"))\n            except StopIteration:\n                raise ValueError(exc[\"message\"]) from None\n    \n            self.txid: str = txid\n            self.source: str = \"\"\n            self.revert_type: str = data[\"error\"]\n            self.pc: Optional[str] = data.get(\"program_counter\")\n            if self.pc and self.revert_type == \"revert\":\n                self.pc -= 1\n    \n            self.revert_msg: Optional[str] = data.get(\"reason\")\n            self.dev_revert_msg = brownie.project.build._get_dev_revert(self.pc)\n            if self.revert_msg is None and self.revert_type in (\"revert\", \"invalid opcode\"):\n                self.revert_msg = self.dev_revert_msg\n            elif self.revert_msg == \"Failed assertion\":\n                self.revert_msg = self.dev_revert_msg or self.revert_msg\n    \n        else:\n>           raise ValueError(str(exc)) from None\nE           ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f116f627520>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py:121: ValueError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "configure_neon_network",
    "time" : {
      "start" : 1642073985866,
      "stop" : 1642073985866,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "brownie_project",
    "time" : {
      "start" : 1642073987038,
      "stop" : 1642073998129,
      "duration" : 11091
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "accounts",
    "time" : {
      "start" : 1642073998589,
      "stop" : 1642073998857,
      "duration" : 268
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "setup_env",
    "time" : {
      "start" : 1642073998589,
      "stop" : 1642073998589,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "token",
    "time" : {
      "start" : 1642074000839,
      "stop" : 1642074001128,
      "duration" : 289
    },
    "status" : "broken",
    "statusMessage" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f116f627520>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.\n",
    "statusTrace" : "  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 1126, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 932, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/neon-compatibility/neon-compatibility/tests/brownie/token/tests/conftest.py\", line 49, in token\n    return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py\", line 600, in __call__\n    return tx[\"from\"].deploy(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 508, in deploy\n    receipt, exc = self._make_transaction(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 725, in _make_transaction\n    raise VirtualMachineError(e) from None\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py\", line 121, in __init__\n    raise ValueError(str(exc)) from None\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  } ],
  "afterStages" : [ {
    "name" : "accounts::0",
    "time" : {
      "start" : 1642074028414,
      "stop" : 1642074028414,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Brownie"
  }, {
    "name" : "parentSuite",
    "value" : "tests.brownie.token.tests"
  }, {
    "name" : "suite",
    "value" : "test_transferFrom"
  }, {
    "name" : "host",
    "value" : "fv-az128-463"
  }, {
    "name" : "thread",
    "value" : "3767-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.brownie.token.tests.test_transferFrom"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 95,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 95
      },
      "items" : [ {
        "uid" : "c1f52bf8fbdce429",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/483//#testresult/c1f52bf8fbdce429",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1642059480215,
          "stop" : 1642059480422,
          "duration" : 207
        }
      }, {
        "uid" : "ec616805e8195152",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/482//#testresult/ec616805e8195152",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1642043778012,
          "stop" : 1642043778154,
          "duration" : 142
        }
      }, {
        "uid" : "53bed225bd1bc0a4",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/481//#testresult/53bed225bd1bc0a4",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1642034888009,
          "stop" : 1642034888199,
          "duration" : 190
        }
      }, {
        "uid" : "1df9f559b73ce34c",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/479//#testresult/1df9f559b73ce34c",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1642033047850,
          "stop" : 1642033048038,
          "duration" : 188
        }
      }, {
        "uid" : "b072bf7b4d73bf4b",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/478//#testresult/b072bf7b4d73bf4b",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: http://proxy.night.stand.neontest.xyz/solana",
        "time" : {
          "start" : 1642032075543,
          "stop" : 1642032075543,
          "duration" : 0
        }
      }, {
        "uid" : "5e492aee68b6d20c",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/477//#testresult/5e492aee68b6d20c",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642031189922,
          "stop" : 1642031189922,
          "duration" : 0
        }
      }, {
        "uid" : "56147c81eb80332e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/476//#testresult/56147c81eb80332e",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642028351504,
          "stop" : 1642028351504,
          "duration" : 0
        }
      }, {
        "uid" : "b63c7707678bcbaf",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/474//#testresult/b63c7707678bcbaf",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642025673657,
          "stop" : 1642025673657,
          "duration" : 0
        }
      }, {
        "uid" : "84d94131d959e0af",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/472//#testresult/84d94131d959e0af",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642018239481,
          "stop" : 1642018239481,
          "duration" : 0
        }
      }, {
        "uid" : "396dc3d650030cb0",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/471//#testresult/396dc3d650030cb0",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642015605310,
          "stop" : 1642015605310,
          "duration" : 0
        }
      }, {
        "uid" : "aa084cf28a74e0d6",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/470//#testresult/aa084cf28a74e0d6",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642014231438,
          "stop" : 1642014231438,
          "duration" : 0
        }
      }, {
        "uid" : "c983f51cfcd99dde",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/469//#testresult/c983f51cfcd99dde",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642012912911,
          "stop" : 1642012912911,
          "duration" : 0
        }
      }, {
        "uid" : "c8f69fa4fd79e3ea",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/468//#testresult/c8f69fa4fd79e3ea",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642011423786,
          "stop" : 1642011423786,
          "duration" : 0
        }
      }, {
        "uid" : "8470ab4acf9689c7",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/466//#testresult/8470ab4acf9689c7",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641998035856,
          "stop" : 1641998036083,
          "duration" : 227
        }
      }, {
        "uid" : "5648fe830d9a1144",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/465//#testresult/5648fe830d9a1144",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641993797374,
          "stop" : 1641993797588,
          "duration" : 214
        }
      }, {
        "uid" : "228fdeb14494b3d7",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/464//#testresult/228fdeb14494b3d7",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641991050878,
          "stop" : 1641991051069,
          "duration" : 191
        }
      }, {
        "uid" : "94be5ec4ea4aeadf",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/463//#testresult/94be5ec4ea4aeadf",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641986395730,
          "stop" : 1641986395871,
          "duration" : 141
        }
      }, {
        "uid" : "c6cf531b86d04bea",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/461//#testresult/c6cf531b86d04bea",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641982479214,
          "stop" : 1641982479214,
          "duration" : 0
        }
      }, {
        "uid" : "686592d6fd201afd",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/460//#testresult/686592d6fd201afd",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189460db20>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641980268252,
          "stop" : 1641980268252,
          "duration" : 0
        }
      }, {
        "uid" : "ec4566601bb4db29",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/459//#testresult/ec4566601bb4db29",
        "status" : "broken",
        "statusDetails" : "OverflowError: approve '-1' - -1 is outside allowable range for uint256",
        "time" : {
          "start" : 1641975867958,
          "stop" : 1641975868158,
          "duration" : 200
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "411682f0177cfaf8.json",
  "parameterValues" : [ ]
}